src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1027,6 +1036,23 @@ public async Task<NuGetProject> UpgradeProjectToPackageReferenceAsync(NuGetProje
             return nuGetProject;
         }
 
+        private void OnSolutionOpened(object sender, EventArgs e)
```

Comment:
You don't necessarily need these. 

These events are raised by this class, so OnSolutionExistsAndFullyLoadedAsync is the solution open equivalent. 
There's a solution closed equivalent.

Reply:
@nkolev92 
I think you're working on stale version. Can you refresh your browser?
I'm not making any changes to this file now.

Reply:
Yes, GH told me. https://github.com/NuGet/NuGet.Client/pull/3755#issuecomment-764168252.

Reply:
@nkolev92 
Are you talking about my change in NuGetpackage.cs?

Reply:
no, ignore this thread. As you said it's comments on an outdated file.

Reply:
@nkolev92 

I added back solution open/close logic here. Somehow solution counter was not working correctly if I hook somewhere else. Please let me know if I need to improve something here.

Reply:
@nkolev92 

After another look, I moved logic into OnSolutionExistsAndFullyLoadedAsync and close equivalent.